Cpp/RequestResponse/src/MAC.lf

```diff
@@ -5,94 +5,105 @@ import MultiplyService from "MultiplyService.lf"
 import ContextManager from "ContextManager.lf"
 
 public preamble {=
-    struct MACData {
-        int factor1;
-        int factor2;
-        int summand;
-    };
+  struct MACData {
+      int factor1;
+      int factor2;
+      int summand;
+  };
 =}
 
 reactor MACService {
-
-    input request: Request<MACData>
-    output response: Response<int>
-
-    output mul_request: Request<std::pair<int, int>>
-    input mul_response: Response<int>
-
-    output add_request: Request<std::pair<int, int>>
-    input add_response: Response<int>
-
-    add_cm = new ContextManager<{=Request<std::pair<int, int>>=}, {=Response<int>=}, {=Request<MACData>=}>()
-    mul_cm = new ContextManager<{=Request<std::pair<int, int>>=}, {=Response<int>=}, {=Request<MACData>=}>()
-
-    reaction(request) -> mul_cm.request_in {=
-        auto& original_request = *request.get();
-        auto& data = original_request.data();
-        auto multiply = std::make_pair(data.factor1, data.factor2);
-        mul_cm.request_in.set(std::make_pair(Request(multiply), original_request));
-    =}
-
-    reaction(mul_cm.response_out) -> add_cm.request_in {=
-        auto mul_result = mul_cm.response_out.get()->first.data();
-        auto& original_request = mul_cm.response_out.get()->second;
-        auto add = std::make_pair(mul_result, original_request.data().summand);
-        add_cm.request_in.set(std::make_pair(Request(add), original_request));
-    =}
-
-    reaction(add_cm.response_out) -> response {=
-        auto result = add_cm.response_out.get()->first.data();
-        auto& original_request = add_cm.response_out.get()->second;
-        response.set(original_request.make_response(result));
-    =}
-
-    mul_cm.request_out -> mul_request
-    add_cm.request_out -> add_request
-    mul_response -> mul_cm.response_in
-    add_response -> add_cm.response_in
+  input request: Request<MACData>
+  output response: Response<int>
+
+  output mul_request: Request<std::pair<int, int>>
+  input mul_response: Response<int>
+
+  output add_request: Request<std::pair<int, int>>
+  input add_response: Response<int>
+
+  add_cm = new ContextManager<
+    {= Request<std::pair<int, int>> =},
+    {= Response<int> =},
+    {= Request<MACData> =}
+  >(
+
```

Comment:
This is an odd formatting decision.

Reply:
What exactly is odd about this? It is common to format lists that are surrounded by parentheses and curly braces in this way. For example, people often write
```
result = function_call(
  first_long_long_argument,
  second_long_long_argument,
  third_long_long_argument,
  fourth_long_long_argument
);
```
and
```
def my_function(arg0, arg1) {
  statement;
  statement;
  ...
}
```

and I do not see how angle brackets are different. How else would this be formatted, if it does not all fit on one line?

Reply:
Sorry, I should have made this clearer. Yes, I think the last line should be `>()`. Or maybe the closing `>` shouldn't even do this. O just tried it put with clang-format using the Google style and it puts the `>()` just after the last type argument (on the same line). Interestingly, clang-format also does not put each type argument on a separate line. It puts as many arguments on the same line until as fit within one line and only then inserts a line break.

Reply:
The issue with the newline inside the parentheses is fixed by https://github.com/lf-lang/lingua-franca/pull/1850.

As for the location of the closing `>` -- I regard this as a matter of opinion. For example, some people like to write functions like the following:
```c
size_t function_call(
  int first_long_long_argument,
  int second_long_long_argument,
  int third_long_long_argument,
  int fourth_long_long_argument
) {
  int a = 2;
  int b;
  ...
}
```
because the `) {` on its own line clearly separates the first list (the parameters) from the second list (the statements in the function body). Like you say, it works just fine to instead do something like

```c
size_t function_call(
    int first_long_long_argument,
    int second_long_long_argument,
    int third_long_long_argument,
    int fourth_long_long_argument) {
  int a = 2;
  int b;
  ...
}
```
where whitespace can be used instead of a line with the closing and opening delimiters as a way to distinguish the two lists.

Both approaches are easy to implement, and both approaches are in widespread use. I'm fine either way.

The idea of putting as much as possible on the same line is, like you say, also done. Black does not do it that way. We can do it that way, but it might be a little harder to implement since it doesn't fit in with our current optimization approach.

Reply:
I don't have a strong opinion about it. I only brought this up because we said we'll try to stay close to the Google style (that is what I understood at least), but apparently the Google style for C++ handles this differently. I am not whether they handle this consistently across languages though.

Let's keep things simple for now and put `) {` or `> ()` on a new line.